File: src/main/java/com/emirhan/ft_case/controller/ProductController.java
package com.emirhan.ft_case.controller;

import com.emirhan.ft_case.domain.common.BaseResponse;
import com.emirhan.ft_case.domain.common.Status;
import com.emirhan.ft_case.domain.model.Product;
import com.emirhan.ft_case.domain.service.ProductService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequiredArgsConstructor
public class ProductController {
    private final ProductService productService;

    @PostMapping("/product")
    ResponseEntity<BaseResponse<Product>> addProduct(@RequestBody Product product) {
        productService.addProduct(product);
        return new ResponseEntity<>(new BaseResponse<>(Status.Success.name, product, null), HttpStatus.OK);
    }

    @GetMapping("/product/expired")
    ResponseEntity<BaseResponse<List<Product>>> getExpiredProducts() {
        return new ResponseEntity<>(new BaseResponse<>(Status.Success.name, productService.getExpiredProducts(), null), HttpStatus.OK);
    }

    @GetMapping("/product/notExpired")
    ResponseEntity<BaseResponse<List<Product>>> getNotExpiredProducts() {
        return new ResponseEntity<>(new BaseResponse<>(Status.Success.name, productService.getNotExpiredProducts(), null), HttpStatus.OK);
    }
}

File: src/main/kotlin/com/emirhan/ft_case/controller/ProductController.kt
package com.emirhan.ft_case.controller

import com.emirhan.ft_case.domain.entity.BaseResponse
import com.emirhan.ft_case.domain.entity.ProductEntity
import com.emirhan.ft_case.domain.entity.Status
import com.emirhan.ft_case.domain.repository.ProductRepository
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class ProductController(private val productRepository: ProductRepository) {
    @PostMapping("/product")
    fun addProduct(@RequestBody product: ProductEntity): ResponseEntity<BaseResponse<Any>> {
        productRepository.addProduct(product)
        return ResponseEntity(BaseResponse(Status.Success.message), HttpStatus.OK)
    }

    @GetMapping("/product/expired")
    fun getExpiredProducts(): ResponseEntity<BaseResponse<List<ProductEntity>>> {
        val productList = productRepository.getExpiredProducts()
        return ResponseEntity(BaseResponse(Status.Success.message, productList), HttpStatus.OK)
    }

    @GetMapping("/product/notExpired")
    fun getNotExpiredProducts(): ResponseEntity<BaseResponse<List<ProductEntity>>> {
        val productList = productRepository.getNotExpiredProducts()
        return ResponseEntity(BaseResponse(Status.Success.message, productList), HttpStatus.OK)
    }
}